graph/tree_diameter.cpp

```diff
@@ -0,0 +1,50 @@
+/**
+ * \file 
+ * \brief  finds the diameter of a tree
+ * \details
+ *  Tree diameter is the longest path between any 2 nodes in the tree
+ * [Tree Diameter](https://en.wikipedia.org/wiki/Treewidth)
+ * 
+ * @author [Rakshit P](https://github.com/rakki-18)
+ */
+ 
+#include<algorithm>
+#include<iostream>
+#include<vector>
+
+std::vector<std::vector<int>>graph;
+int tree_diameter;
```

Comment:
Don't create global variables, define a function which takes all the input values as parameters and returns the output as return value.